### DISCUSSION TITLE
SUN-29.9/30/35/40/50K-SG01HP3-EU

### DISCUSSION THREAD

**Discussion by JanKraslice** [StephanJoubert/home_assistant_solarman#443] [General]
Hello, is it possible to apply for integration ?  
SUN-29.9/30/35/40/50K-SG01HP3-EU

This is the HV Battery version, which is already quite widespread compared to the LV Battery version 
The modbus is identical but with the settings for 8,10,12K (deye_sg04lp3.yaml) it does not display correctly for example the battery voltage 

Thank you

**Comment by JanKraslice** [comment]
This is DEYE SUN-29.9/30/35/40/50K-SG01HP3-EU

**Reply by JanPova** [reply]
i have same problem :(

**Comment by Z3tH1024** [comment]
i got the SUN-8K-SG01HP3-EU-AM2.
It works for me with the Template deye_sg04lp3 but i dont find my actually right now produce Solarpower.

**Reply by TomFeix** [reply]
you can edit the file for more information

**Reply by JanKraslice** [reply]
Can you please tell me how to edit the file? 
I mean exactly what file.
Example if I can porposti
thank you

**Reply by TomFeix** [reply]
/homeassistant/custom_components/solarman/inverter_definitions/deye_sg04lp3.yaml this is the file for my inverter
and there is a description on deye
[MODBUS RTU三相储能通信规约V104（高压）.pdf](https://github.com/StephanJoubert/home_assistant_solarman/files/15432010/MODBUS.RTU.V104.pdf)

**Reply by JanKraslice** [reply]
Thank you, but the file you sent me is available. 
Rather, I need advice on how to add the missing for example
battery_max_charge_current
or
pv2_power

to which file or how to edit it

Thank you

**Reply by TomFeix** [reply]
I sent you my yaml
[HA_Config.txt](https://github.com/user-attachments/files/15523053/HA_Config.txt)
depens on the template of your model
